### PR TITLE
refactor: small cleanup around realms and handles

### PR DESCRIPTION
**Pull request by jrandolf-google**

### CODE REVIEW

**Review by OrKoN — CHANGES_REQUESTED** [review]
tests are failing

File: packages/puppeteer-core/src/bidi/Deserializer.ts [hunk 1]
```diff
@@ -27,25 +27,30 @@ export class BidiDeserializer {
     }
   }
 
-  static deserializeLocalValue(result: Bidi.Script.RemoteValue): unknown {
+  static deserializeRemoteValue(result: Bidi.Script.RemoteValue): any {
```
Comment (Lightning00Blade) [inline packages/puppeteer-core/src/bidi/Deserializer.ts]:
```suggestion
  static deserialize(result: Bidi.Script.RemoteValue): any {
```

Comment (jrandolf) [reply packages/puppeteer-core/src/bidi/Deserializer.ts]: I changed it to deserializeRemoteValue to mirror Serializer.

File: packages/puppeteer-core/src/bidi/Realm.ts [hunk 2]
```diff
@@ -203,10 +203,29 @@ export class BidiRealm extends EventEmitter<Record<EventType, any>> {
     }
 
     return returnByValue
-      ? BidiDeserializer.deserialize(result.result)
+      ? BidiDeserializer.deserializeRemoteValue(result.result)
       : createBidiHandle(sandbox, result.result);
   }
 
+  async destroyHandles(handles: Array<BidiJSHandle<unknown>>): Promise<void> {
```
Comment (Lightning00Blade) [inline packages/puppeteer-core/src/bidi/Realm.ts:210]: Nit: Let's not send command if handles is empty after filtering?

File: packages/puppeteer-core/src/bidi/Deserializer.ts (hunk 1 above)
Comment (Lightning00Blade) [reply packages/puppeteer-core/src/bidi/Deserializer.ts]: I think we should keep simple names, as now we are duplicating the type in the function name without having another function. So we should change both places.

**Review by Lightning00Blade — APPROVED** [review]
LGTM

**Pull request merged**